lib/Site.js

```diff
@@ -234,49 +266,68 @@ Site.prototype.rebuildSourceFiles = function (filePath) {
   });
 };
 
-Site.prototype.buildAsset = function (filePath) {
+/**
+ * Rebuild pages that are affected by change in filePaths
+ * @param filePaths paths corresponding to file that have changed
+ */
+Site.prototype.rebuildSourceFiles
+  = delay(Site.prototype._rebuildSourceFiles, 1000);
+
+Site.prototype._buildMultipleAssets = function (filePaths) {
+  const uniquePaths = _.uniq(filePaths);
+  const ignoreConfig = this.siteConfig.ignore || [];
+  const fileIgnore = ignore().add(ignoreConfig);
+  const relativeFiles = uniquePaths.map(filePath => path.relative(this.rootPath, filePath));
+  const copyAssets = fileIgnore.filter(relativeFiles)
+    .map(asset => fs.copyAsync(path.join(this.rootPath, asset), path.join(this.outputPath, asset)));
   return new Promise((resolve, reject) => {
-    // if the file is an ignored file, resolve
-    // Else, copy it to its destination
-    const ignoreConfig = this.siteConfig.ignore || [];
-    const fileRelative = path.relative(this.rootPath, filePath);
-    const fileIgnore = ignore().add(ignoreConfig);
-    if (fileIgnore.filter([fileRelative]).length === 0) {
-      resolve();
-    } else {
-      fs.copyAsync(filePath, path.join(this.outputPath, fileRelative))
-        .then(resolve)
-        .catch((error) => {
-          rejectHandler(reject, error, []); // assets won't affect deletion
-        });
-    }
+    Promise.all(copyAssets)
+      .then(() => logger.info('Assets built'))
+      .then(resolve)
+      .catch((error) => {
+        rejectHandler(reject, error, []); // assets won't affect deletion
+      });
   });
 };
 
-Site.prototype.removeAsset = function (filePath) {
+/**
+ * Build/copy assets that are specified in filePaths
+ * @param filePaths paths corresponding to the assets to build
+ */
+Site.prototype.buildAsset
+ = delay(Site.prototype._buildMultipleAssets, 1000);
+
+Site.prototype._removeMultipleAssets = function (filePaths) {
+  const uniquePaths = _.uniq(filePaths);
+  const relativeFiles = uniquePaths.map(filePath => path.relative(this.rootPath, filePath));
+  const filesToRemove = relativeFiles.map(fileRelative => path.join(this.outputPath, fileRelative));
+  const removeFiles = filesToRemove.map(asset => fs.removeAsync(asset));
   return new Promise((resolve, reject) => {
-    const fileRelative = path.relative(this.rootPath, filePath);
-    const fileToRemove = path.join(this.outputPath, fileRelative);
-    fs.removeAsync(fileToRemove)
+    Promise.all(removeFiles)
```

Comment:
- Remove explicit promise construction when using `Promise.all`.
- Do the same in `Site.prototype.buildAllAssets`.